subsys/bluetooth/mesh/adv.h

```diff
@@ -80,6 +80,8 @@ int bt_mesh_scan_disable(void);
 
 int bt_mesh_adv_enable(void);
 
+int bt_mesh_adv_disable(void);
```

Comment:
Please add it shouldn't be called from work queue since behavior is undefined.
Probably it is worth to add to public suspension API

This is `k_work_flush_delayable` note:
```
Note
Be careful of caller and work queue thread relative priority. If this function sleeps it will not return until the work queue thread completes the tasks that allow this thread to resume.
Behavior is undefined if this function is invoked on dwork from a work queue running dwork.
```

Reply:
Added comment to `adv.h` and a note in `bt_mesh_suspend()` API description